### ISSUE TITLE
admission chain is given authorizer that doesn't include superuser fallback

### ISSUE THREAD

**Issue by jindijamie** [kubernetes/kubernetes#109211]
### What happened?

I created a v1.21.5 cluster. After I deleted the default cluster-admin clusterrolebinding with group "system: master", I failed to approve a pending csr with "kubectl certificate approve csr" and get the following error
Error from server (Forbidden): certificatesigningrequests.certificates.k8s.io "csr-abcdefg" is forbidden: user not permitted to approve requests with signerName "kubernetes.io/kubelet-serving"

### What did you expect to happen?

the clusterrolebinding is not needed for the system:masters group to have superuser permissions, delete the clusterrolebinding shouldn't cause loss of permissions for the system:masters group

### How can we reproduce it (as minimally and precisely as possible)?

- Create a cluster
- kubectl delete clusterrolebinding cluster-admin
- Create a csr in the cluster
- Try to approve that csr with `kubectl certificate approve`

### Kubernetes version

<details>

```console
$ kubectl version
v1.21.5
```

</details>

### Cloud provider

<details>
vmware
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by liggitt** [comment]
I suspect the admission plugin initializers are getting constructed before the wrapping/prefixing of the authorizer with the superuser authorizer

cc @deads2k

**Comment by mikedanese** [comment]
We add that superuser authorizer during config completion but we yank the authorizer for admission in ApplyTo which happens before. We can yank this:

https://github.com/kubernetes/kubernetes/blob/75e49ec824b183288e1dbaccfd7dbe77d89db381/staging/src/k8s.io/apiserver/pkg/server/config.go#L939-L940

Into the authorizer config in kube-apiserver so it happens early. It's kind of a tangled mess.

**Comment by mrjoema** [comment]
Hi all, can I try to take this issue ? Since I am new to k8s, I will need some time to ramp it up if it's okay for you all.

**Comment by liggitt** [comment]
> We can yank this:
> https://github.com/kubernetes/kubernetes/blob/75e49ec824b183288e1dbaccfd7dbe77d89db381/staging/src/k8s.io/apiserver/pkg/server/config.go#L939-L940
> 
> Into the authorizer config in kube-apiserver so it happens early. It's kind of a tangled mess.

We also need to make sure it gets included in the authorizer chain constructed for aggregated API servers

**Comment by liggitt** [comment]
> Hi all, can I try to take this issue ? Since I am new to k8s, I will need some time to ramp it up if it's okay for you all.

I would say this is a medium difficulty fix to implement and test. If you're just starting out, a more contained issue might be a better place to jump in.

That said, if no one else grabs it between now and 1.25 development opening up, and you want to take it on, go for it. We can pair you with someone who knows the area to orient/review (@deads2k, @mikedanese, @enj, or myself are likely candidates)

**Comment by deads2k** [comment]
I suspect you're looking for the code that creates this admission initializer https://github.com/kubernetes/kubernetes/blob/75e49ec824b183288e1dbaccfd7dbe77d89db381/staging/src/k8s.io/apiserver/pkg/admission/initializer/initializer.go#L37-L70 .  That's the style in which admission plugins gain access to authorizers.

**Comment by harshitasao** [comment]
Hi @jindijamie & @liggitt , k8s 1.25 Bug Triage Shadow here, just checking in to see if this is still on track for k8s 1.25?
Thank you.

**Comment by liggitt** [comment]
it's still important to fix, no one has picked it up to do the work yet

**Comment by jlsong01** [comment]
I would like to take a try
/assign

**Comment by harshitasao** [comment]
> it's still important to fix, no one has picked it up to do the work yet

Thanks for the update.

**Comment by ritazh** [comment]
@jlsong01 Are you still working on this issue?

**Comment by jlsong01** [comment]
I am busy recently. Please go ahead and take it if you are insterester 😀

**Comment by mzkhan** [comment]
I would like to take this one. @liggitt Can you assign it to me.

**Comment by liggitt** [comment]
sure, thanks!

/assign @mzkhan

**Comment by harshitasao** [comment]
Hello @liggitt, with the code freeze approaching, is this still on track for v1.25?

**Comment by liggitt** [comment]
> Hello @liggitt, with the code freeze approaching, is this still on track for v1.25?

no work has started on it as far as I'm aware

**Comment by HeshamAboElMagd** [comment]
Hello @mzkhan , just wanted to touch-base for any update regarding this issue. 
Thanks!

**Comment by mzkhan** [comment]
Hi @HeshamAboElMagd , the PR is waiting on me to add tests for the change. I will be working on it ,this week to see it through.

**Comment by HeshamAboElMagd** [comment]
Thanks @mzkhan !
